specs/TransportProtocols.md

```diff
@@ -0,0 +1,176 @@
+* `[Body]` - The body of the message. If `[Type]` is `T`, this is just the sequence of UTF-8 characters for the text frame. If `[Type]` is `B`, the frame is Base64-encoded, so `[Body]` must be Base64-decoded to get the actual frame content.
+
+For example, when sending the following frames (`\n` indicates the actual Line Feed character, not an escape sequence):
+
+* Type=`Text`, Fin=`true`, "Hello\nWorld"
+* Type=`Binary`, Fin=`false`, `0x01 0x02`
+* Type=`Binary`, Fin=`true`, `0x03 0x04`
+
+The encoding will be as follows
+
+```
+T11:T,T:Hello
+World;4:B,F:AQI=;4:B,T:AwQ=;
+```
+
+Note that the final frame still ends with the `;` terminator, and that since the body may contain `;`, newlines, etc., the length is specified in order to know exactly where the body ends.
+
+### Binary encoding (`supportsBinary` = `true`)
+
+In JavaScript/Browser clients, this encoding requires XHR2 (or similar HTTP request functionality which allows binary data) and TypedArray support.
+
+The body is encoded as follows. Identifiers in square brackets `[]` indicate fields defined below, and parenthesis `()` indicate grouping. Other symbols indicate ASCII-encoded text in the stream
+
+```
+B([Length][TypeAndFin][Body])([Length][Type][Fin][Body])... continues until end of the response body ...
+```
+
+* `[Length]` - A 64-bit integer in Network Byte Order (Big-endian) representing the length of the body in bytes
+* `[TypeAndFin]` - An 8-bit integer where the most significant bit indicates if the frame is the last frame in a message (`1` indicates `true`, `0` indicates `false`) and the least significant bit indicates if the frame is text or binary (`0` indicates Text and `1` indicates Binary). Thus, the following four values are the only valid values:
+    * `0x00` - Fin=`false`, Type=`Text`
+    * `0x01` - Fin=`false`, Type=`Binary`
+    * `0x80` - Fin=`true`, Type=`Text`
+    * `0x81` - Fin=`true`, Type=`Binary`
+    * All other bits are reserved and MUST be `0` or the entire response should be rejected.
+* `[Body]` - The body of the message, exactly `[Length]` bytes in length. Text frames are always encoded in UTF-8.
+
+For example, when sending the following frames (`\n` indicates the actual Line Feed character, not an escape sequence):
+
+* Type=`Text`, Fin=`true`, "Hello\nWorld"
+* Type=`Binary`, Fin=`false`, `0x01 0x02`
+* Type=`Binary`, Fin=`true`, `0x03 0x04`
+
+The encoding will be as follows, as a list of binary digits in hex (text in parentheses `()` are comments). Whitespace and newlines are irrelevant and for illustration only.
+```
+0x66                                                   (ASCII 'B')
+0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x0B                (start of frame; 64-bit integer value: 11)
+0x80                                                   (Type = Text, Fin = True)
+0x68 0x65 0x6C 0x6C 0x6F 0x0A 0x77 0x6F 0x72 0x6C 0x64 (UTF-8 encoding of 'Hello\nWorld')
```

Comment:
I looked into this actually and it's not that hard to do. TextDecoder is experimental but at least there's something the browser vendors are looking at. Let's stick with utf8.